generate/src/contract/events.rs

```diff
@@ -42,10 +43,18 @@ fn expand_structs_mod(cx: &Context) -> Result<TokenStream> {
     })
 }
 
+fn expand_derives(derives: &[TypePath]) -> TokenStream {
+    let derives = derives
+        .iter()
+        .map(|derive| derive.to_token_stream())
+        .collect::<Vec<_>>();
+    quote! {#(#derives),*}
```

Comment:
nit: `ToTokens::to_token_stream()` gets called automatically for you by `quote!`.

```suggestion
    quote! {#(#derives),*}
```